### PR TITLE
store sha1 hash of docLines in redis

### PR DESCRIPTION
**Pull request by briangough**
Store and check the sha1 hash of docLines stored in redis

### CODE REVIEW

File: app/coffee/RedisManager.coffee [hunk 1]
```diff
@@ -56,11 +61,19 @@ module.exports = RedisManager =
...
+		multi.get keys.docHash(doc_id:doc_id)
 		multi.get keys.projectKey(doc_id:doc_id)
 		multi.get keys.ranges(doc_id:doc_id)
-		multi.exec (error, [docLines, version, doc_project_id, ranges])->
+		multi.exec (error, [docLines, version, storedHash, doc_project_id, ranges])->
 			timer.done()
 			return callback(error) if error?
+
+			# check sha1 hash value if present
+			if docLines? and storedHash?
```
Comment (jpallen) [inline app/coffee/RedisManager.coffee]: This logic should have some unit tests I think. It's non-trivial enough, and difficult to test manually.

File: app/coffee/RedisManager.coffee [hunk 2]
```diff
@@ -121,8 +134,11 @@ module.exports = RedisManager =
 				return callback(error)
 			jsonOps = appliedOps.map (op) -> JSON.stringify op
 			multi = rclient.multi()
-			multi.set    keys.docLines(doc_id:doc_id), JSON.stringify(docLines)
+			newDocLines = JSON.stringify(docLines)
+			newHash = RedisManager._computeHash(newDocLines)
+			multi.set    keys.docLines(doc_id:doc_id), newDocLines
 			multi.set    keys.docVersion(doc_id:doc_id), newVersion
+			multi.set    keys.docHash(doc_id:doc_id), newHash
```
Comment (jpallen) [inline app/coffee/RedisManager.coffee]: Add some simple unit tests to check that we're setting the hash with the correct value?

File: app/coffee/RedisManager.coffee (hunk 1 above)
Comment (briangough) [inline app/coffee/RedisManager.coffee]: now added

File: app/coffee/RedisManager.coffee (hunk 2 above)
Comment (briangough) [inline app/coffee/RedisManager.coffee]: now added

**Pull request merged**